pkg/operator/ceph/config/keyring/cephx.go

```diff
@@ -0,0 +1,186 @@
+/*
+Copyright 2025 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package keyring
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+	v1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	"github.com/rook/rook/pkg/operator/ceph/version"
+)
+
+var CephAuthRotateSupportedVersion = version.CephVersion{Major: 20, Minor: 2, Extra: 0}
+
+// CephxKeyIdentifierAnnotation is the annotation that should be applied to pod specs to
+// ensure that pods restart after keys are rotated (and not restarted when keys are not rotated).
+// The keyring secret resourceVersion is suggested but not always available.
+//
+//nolint:gosec // G101: this is not hardcoded credentials
+const CephxKeyIdentifierAnnotation = "cephx-key-identifier"
+
+// ShouldRotateCephxKeys determines whether CephX keys should be rotated based on the CephX key
+// rotation config, the version of Ceph present in the image being deployed (desiredCephVersion),
+// and the last-reconciled CephX key status.
+// runningCephVersion is used to determine if the cluster is capable of rotating CephX keys.
+// Intended to use running/desired ceph version from CurrentAndDesiredCephVersion().
+func ShouldRotateCephxKeys(cfg v1.CephxConfig, runningCephVersion, desiredCephVersion version.CephVersion, status v1.CephxStatus) (bool, error) {
+	if !runningCephVersion.IsAtLeast(CephAuthRotateSupportedVersion) {
+		logger.Debugf("should not rotate cephx keys using unsupported ceph version %#v", runningCephVersion)
+		return false, nil
+	}
+
+	if status.KeyCephVersion == v1.UninitializedCephxKeyCephVersion {
+		return false, nil // no need to rotate key when key isn't yet initialized
+	}
+
+	switch cfg.KeyRotationPolicy {
+	case v1.CephxKeyRotationPolicy(""), v1.DisabledCephxKeyRotationPolicy:
+		return false, nil
+	case v1.KeyGenerationCephxKeyRotationPolicy:
+		return cfg.KeyGeneration > status.KeyGeneration, nil
+	case "WithCephVersionUpdate": // TODO: use types.go value when allowed by user input
```

Comment:
Shall we rename this now, per discussion below? Or we can just defer if still pondering the name.
```suggestion
	case "WithCephImageUpdate": // TODO: use types.go value when allowed by user input
```